### PR TITLE
Collect network traffic and use it to extract DNS requests and responses.

### PR DESCRIPTION
**Pull request by calebbrown**
This helps solve issue #44.

Use the libpcap and gopacket library to collect and handle DNS traffic.

Can be extended to handle other types of traffic.

### CODE REVIEW

File: internal/dnsanalyzer/dnsanalyzer.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+// Hostname returns the hostname used to obtain the given IP address.
+//
+// Returns the hostnames found. Otherwise it returns an empty slice.
+func (d *DNSAnalyzer) Hostnames(address string) []string {
+	// We parse the IP to ensure that it is valid.
+	ip := net.ParseIP(address)
+	if ip == nil {
+		return []string{}
```
Comment (oliverchang) [inline internal/dnsanalyzer/dnsanalyzer.go:96]: Should we log or return an error here?

File: internal/dnsanalyzer/dnsanalyzer.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	"github.com/google/gopacket/layers"
+)
+
+type rawLookup struct {
+	query *layers.DNS
+	reply *layers.DNS
+}
+
+type DNSAnalyzer struct {
+	lookups     map[uint16]*rawLookup
```
Comment (oliverchang) [inline internal/dnsanalyzer/dnsanalyzer.go:16]: Do we actually need to save these?

File: internal/dnsanalyzer/dnsanalyzer.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	ipHostnames map[string]map[string]struct{}
```
Comment (oliverchang) [inline internal/dnsanalyzer/dnsanalyzer.go:17]: Let's make a `type empty struct{}` to make this more readable.

**Review by oliverchang — APPROVED** [review]
nice!!

File: internal/dnsanalyzer/dnsanalyzer.go (hunk 3 above)
Comment (calebbrown) [reply internal/dnsanalyzer/dnsanalyzer.go]: Good idea. Done!

File: internal/dnsanalyzer/dnsanalyzer.go (hunk 2 above)
Comment (calebbrown) [reply internal/dnsanalyzer/dnsanalyzer.go]: I was going to use them in the future, but its better to keep unused code out.

File: internal/dnsanalyzer/dnsanalyzer.go (hunk 1 above)
Comment (calebbrown) [reply internal/dnsanalyzer/dnsanalyzer.go:82]: I don't think so. This function is basically a hash lookup. The ParseIP() usage here is merely to ensure the IP is normalized.

Comment (calebbrown) [reply internal/dnsanalyzer/dnsanalyzer.go:83]: I updated the comment to make this clearer.

**Review by oliverchang — APPROVED** [review]

**Pull request merged**